ci-operator/step-registry/aws/provision/route53/private-hosted-zone/associate-with-vpc-in-different-account/aws-provision-route53-private-hosted-zone-associate-with-vpc-in-different-account-commands.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+set -x
+
+REGION="${LEASED_RESOURCE}"
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+if [[ ${BASE_DOMAIN} == "" ]]; then
+  echo "No base_domain provided, exit now."
+  exit 1
+fi
+ROUTE53_HOSTED_ZONE_NAME="${CLUSTER_NAME}.${BASE_DOMAIN}"
+
+VPC_ID=$(cat "${SHARED_DIR}/vpc_id")
+# Use a timestamp to ensure the caller reference is unique, as we've found
+# cluster name can get reused in specific situations.
+TIMESTAMP=$(date +%s)
+CALLER_REFERENCE_STR="${ROUTE53_HOSTED_ZONE_NAME}-${TIMESTAMP}"
+
+
+# Create PHZ
+echo "AWS Account: using shared account"
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred_shared_account"
```

Comment:
Per the logic, maybe we can name the step to `aws/provision/route53/private-hosted-zone/shared-associate-vpc-account/aws-provision-route53-private-hosted-zone-shared-associate-vpc-account`

Reply:
How about `associate-diff-vpc`

Reply:
In this case, we are provisioning phz in shared account, associate it to a vpc in a different account, on the opposite, I guess we also can have a case like `provisioning phz in a common account, associate it to a vpc in a shared account`, unless we think the two accounts make no difference.

Reply:
Technically I think OCP supports that, it's similar to BYO-PHZ configuration: once VPC is shared with cluster (PHZ) account, from OCP's perspective, it's almost the same as BYO-VPC,  we don't need to operate PHZ belongs to another account (No `hostedZoneRole` required).  Step `aws-provision-route53-private-hosted-zone` should be used.

But I don't think we need to add such configuration in CI, as I don't see any stories/doc or user request for that. Shared-VPC (+PHZ) is officially supported and Shared-PHZ configuration comes from an SE ticket. WDYT?